src/block/NetherVines.php

```diff
@@ -0,0 +1,178 @@
+use pocketmine\entity\Entity;
+use pocketmine\event\block\StructureGrowEvent;
+use pocketmine\item\Fertilizer;
+use pocketmine\item\Item;
+use pocketmine\item\Shears;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+use pocketmine\world\BlockTransaction;
+use function min;
+use function mt_rand;
+
+/**
+ * This class is used for Weeping & Twisting vines, because they have same behaviour
+ */
+abstract class NetherVines extends Flowable{
+	public const MAX_AGE = 25;
+
+	protected int $age = 0;
+
+	public function getRequiredStateDataBits() : int{
+		return 5;
+	}
+
+	public function describeState(RuntimeDataWriter|RuntimeDataReader $w) : void{
+		$w->boundedInt(5, 0, self::MAX_AGE, $this->age);
+	}
+
+	public function getAge() : int{
+		return $this->age;
+	}
+
+	/** @return $this */
+	public function setAge(int $age) : self{
+		if($age < 0 || $age > self::MAX_AGE){
+			throw new \InvalidArgumentException("Age must be in range 0-" . self::MAX_AGE);
+		}
+
+		$this->age = $age;
+		return $this;
+	}
+
+	public function isAffectedBySilkTouch() : bool{
+		return true;
+	}
+
+	public function ticksRandomly() : bool{
+		return true;
+	}
+
+	public function canClimb() : bool{
+		return true;
+	}
+
+	abstract protected function getGrowthFace() : int;
+
+	protected function getSupportFace() : int{
+		return Facing::opposite($this->getGrowthFace());
+	}
+
+	private function canBeSupportedBy(Block $block) : bool{
+		return $block->getSupportType($this->getSupportFace())->hasCenterSupport() || $block->isSameType($this);
+	}
+
+	public function onNearbyBlockChange() : void{
+		if(!$this->canBeSupportedBy($this->getSide($this->getSupportFace()))){
+			$this->position->getWorld()->useBreakOn($this->position);
+		}
+	}
+
+	private function seekToTop() : NetherVines{
+		$face = $this->getGrowthFace();
+		$top = $this;
+		while(($next = $top->getSide($face))->isSameType($this)){
+			$top = $next;
+		}
+		return $top;
+	}
+
+	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if(!$this->canBeSupportedBy(($support = $blockReplace->getSide($this->getSupportFace())))){
+			return false;
+		}
+		$this->age = $support->isSameType($this) ? min($support->getAge() + 1, self::MAX_AGE) : mt_rand(0, self::MAX_AGE - 1);
+		return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
+	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if($item instanceof Fertilizer){
+			if($this->seekToTop()->getSide($this->getGrowthFace())->canBeReplaced()){
+				if($this->grow($player, mt_rand(1, 5))){
+					$item->pop();
+				}
+			}
+			return true;
+		}
+		return false;
+	}
+
+	public function onRandomTick() : void{
+		if(mt_rand(0, 100) <= 10 && $this->age < self::MAX_AGE){
```

Comment:
This should be 0,99 if you're trying to make a 1/100 chance, and it should be <10 if you want a 10% chance.